crud/src/test/java/com/redhat/lightblue/assoc/RelativePathTest.java

```diff
@@ -0,0 +1,36 @@
+/*
+ Copyright 2013 Red Hat, Inc. and/or its affiliates.
+
+ This file is part of lightblue.
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.redhat.lightblue.assoc;
+
+import org.junit.Test;
+import org.junit.Assert;
+
+import com.redhat.lightblue.util.Path;
+
+public class RelativePathTest {
+    @Test
+    public void testRelativePaths() throws Exception {
```

Comment:
Added test cases where cmdPrefix is nonempty.

Added some comments to the function itself. The path field cannot be
different from fullPath for this function, it gets the fullpath to field,
the field as it appears in the expression, and the entity prefix containing
the field, and writes an entity relative field name, so a suffix of the
field name has to match a suffix of the fullpath.

On Fri, Oct 2, 2015 at 6:52 AM, Naveen Malik <notifications@github.com>
wrote:

> In crud/src/test/java/com/redhat/lightblue/assoc/RelativePathTest.java
> <https://github.com/lightblue-platform/lightblue-core/pull/481#discussion_r41017655>
> :
>
> > + MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
> > + GNU General Public License for more details.
> > +
> > + You should have received a copy of the GNU General Public License
> > + along with this program.  If not, see <http://www.gnu.org/licenses/>.
> > + */
> > +package com.redhat.lightblue.assoc;
> > +
> > +import org.junit.Test;
> > +import org.junit.Assert;
> > +
> > +import com.redhat.lightblue.util.Path;
> > +
> > +public class RelativePathTest {
> > +    @Test
> > +    public void testRelativePaths() throws Exception {
>
> Would like to see path have field different from fullPath.
> And add tests with cmdPrefix non-empty.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/481/files#r41017655>
> .
>